docs/metrics/average-precision.md

```diff
@@ -0,0 +1,261 @@
+with low APs due to the approximation mentioned above. The [COCO benchmark](https://cocodataset.org/#detection-eval)
+uses a linear interpolation method but with **101** recall values.
+
+### Information Retrieval
+
+Information retrieval is a machine learning workflow where the user provides a query, and the model returns a score
+that measures how similar each data is to the query to find the most relevant information from the database.
+
+Average precision is one of the popular metrics used in information retrieval workflow along with object detection
+workflow. However, the implementation used in information retrieval workflow is slightly different than the one from
+the section above. The formula of the metric is defined:
+
+$$
+\text{AP}(n) = \frac 1 {\text{GTP}} \sum_k^{n}p(k) \times rel(k)
+$$
+
+where
+
+- $n$ is the total number of data that you are interested in
+- $\text{GTP}$ is the total number of positive ground truths
+- $p(k)$ is the precision at rank $k$ data
+- $rel(k)$ is the relevance at rank $k$ data (1 if the data is relevant, 0 otherwise)
+
+
+Let’s consider the following example of retrieving similar images to the query from a database of images with different
+shapes and colors:
+
+![information retrieval example](../assets/images/metrics-ap-ir-example.png)
+
+The retrieved images are the complete list of images from the database that are ranked by their similarity scores,
+which are predicted from the model, where the left-most image is the most similar to the query image.
+
+From the retrieved images, the ones with a circle are the TPs, where $rel(k) = 1$, and any other shapes are labeled as
+FPs, where $rel(k) = 0$. Then by simply accumulating all the counts of the TPs in each rank, $p(k) \times rel(k)$ can be
+calculated at each rank.
+
+AP is the sum of all the relevant precisions over the total number of positive samples in the database, so in this
+example, AP becomes:
+
+$$
+\text{AP} = \frac {(\frac 1 1 + \frac 0 2 + \frac 0 3 + \frac 2 4 + \frac 3 5 + \frac 0 6 + ... + 0)} 3 = 0.7
+$$
+
+### mean Average Precision (mAP)
+
+The mean average precision (mAP) is simply the [macro-average](./averaging-methods.md) of the AP calculated across
+different classes for object detection workflow or across different queries for information retrieval workflow. It is
+important to note that some papers use AP and mAP interchangeably.
+
+## Limitations and Biases
+
+AP is a great metric that summarizes the PR curve into a single value. Instead of comparing models with a single value
+of precision and recall at one specific threshold, it lets you compare model performance at every threshold. Although
+this metric is very popular and commonly used in object detection and information retrieval workflows, it has some
+limitations. Let's make sure to understand these limitations before using the metric to compare your models.
+
+1. AP is often overestimated.
```

Comment:
I like these limitations a lot. Can we make them pop more? Maybe bolding the header, or making each into a callout (using [`info`](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#type:info), maybe)?

Reply:
Tried the callout.  Let me know if you think this is too much
![Uploading Screen Shot 2023-07-11 at 4.32.08 PM.png…]()

Reply:
What do you think about making them collapsed (and `info`) by default? Up to you — reverting to a numbered list would also be fine.

![Screen Shot 2023-07-11 at 4 37 09 PM](https://github.com/kolenaIO/kolena/assets/9648886/3a55cc70-8423-4a45-96be-10ca61e4fe40)

Reply:
I'm good with the collapsible info.  I will make this change in a separate PR quickly.